src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -334,6 +340,57 @@ private StudentExam findStudentExamWithExercisesElseThrow(User user, Long examId
         return studentExam;
     }
 
+    /**
+     * GET /courses/{courseId}/exams/{examId}/student-exams/grade-summary : Return student exam result, aggregate points, assessment result
+     * for a student exam and grade calculations if the exam is assessed. Only instructors can use userId parameter to get exam results of other users,
+     * if the caller is a student, userId should either be the user id of the caller or empty.
+     *
+     * Does not return the student exam itself to save bandwidth.
+     *
+     * See {@link StudentExamWithGradeDTO} for more explanation.
+     *
+     * @param courseId  the course to which the student exam belongs to
+     * @param examId    the exam to which the student exam belongs to
+     * @param userId    the user id of the student whose grade summary is requested
+     * @return the ResponseEntity with status 200 (OK) and with the StudentExamWithGradeDTO instance without the student exam as body
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/student-exams/grade-summary")
```

Comment:
Is there a test for this particular endpoint? I am not sure if I missed it :thinking:

Reply:
Yes, there are multiple tests for that endpoint in [StudentExamIntegrationTest.java](https://github.com/ls1intum/Artemis/pull/4932/files#diff-429a981130449a3a736220db07846e1cded37f11d26ad03f5f4224c4eeaec595), GitHub collapses the large diff by default so you may have missed it if you tried searching for it with `Ctrl+F` 😄